### PR TITLE
new: `file_server.sort`

### PR DESCRIPTION
**Pull request by lollipopkit**

### CODE REVIEW

**Comment by lollipopkit** [comment]
I'm sorry for opening this PR before the related [PR](https://github.com/caddyserver/caddy/pull/6468) is accepted.

File: src/docs/markdown/caddyfile/directives/file_server.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ file_server [<matcher>] [browse] {
 	status        <status>
 	disable_canonical_uris
 	pass_thru
+	sort          <options...>
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/file_server.md]:
```suggestion
	sort          <sort_field> [<direction>]
```

Comment (mholt) [reply src/docs/markdown/caddyfile/directives/file_server.md]: This also needs to be moved to within the `browse` subdirective block above

File: src/docs/markdown/caddyfile/directives/file_server.md [hunk 2]
```diff
@@ -69,6 +70,9 @@ file_server [<matcher>] [browse] {
 
 - **pass_thru** <span id="pass_thru"/> enables pass-thru mode, which continues to the next HTTP handler in the route if the requested file is not found, instead of triggering a `404` error (invoking [`handle_errors`](handle_errors) routes). Practically, this is only useful inside of a [`route`](route) block with other handler directives following `file_server`, because this directive is effectively [ordered last](/docs/caddyfile/directives#directive-order).
 
+- **sort** <span id="sort"/> specifies how to sort directory listings. The options are sortBy (`name`, `namedirfirst`, `size`, `time`) and order (`asc` or `desc`) separated by spaces. The first option must be `sortBy`, and the second option must be `order` (if exists). For example, `sort name desc` will sort by name in descending order.
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/file_server.md]:
```suggestion
- **sort** <span id="sort"/> changes the default sort for directory listings. The first parameter is the field/column to sort by: `name`, `namedirfirst`, `size`, or `time`. The second argument is an optional direction: `asc` or `desc`. For example, `sort name desc` will sort by name in descending order.
```

Comment (mholt) [reply src/docs/markdown/caddyfile/directives/file_server.md]: This also needs to be moved up into the sub-list of browse.

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for the docs! Just a few changes needed I think.

File: src/docs/markdown/caddyfile/directives/file_server.md (hunk 2 above)
Comment (lollipopkit) [reply src/docs/markdown/caddyfile/directives/file_server.md]: Moved to `browse`, but the location of `sort` config...

It seems I need to change the caddy code again.

```
file_server browse {
	sort size desc
}
```

```
file_server {
	browse {
		sort size desc
	}
}
```
Change 1 to 2? or make both valid, but 2 has higher priority? or just keep current?

Comment (mholt) [reply src/docs/markdown/caddyfile/directives/file_server.md]: Good question; I think the first one is not really a thing; i.e. it's just a shorthand for the default browse (no block).

So aim for the second one. Just put it inside the existing browse block. :+1:

**Comment by francislavoie** [comment]
Merging into the 2.9 branch. Thanks!

**Pull request merged**